operator/pkg/ciliumidentity/reconciler.go

```diff
@@ -163,6 +160,10 @@ func (r *reconciler) reconcileCID(cidResourceKey resource.Key) error {
 			return r.createCID(cidName, cidKey)
 		} else {
 			r.desiredCIDState.Remove(cidName)
+			err := r.makeIDAvailable(cidName)
+			r.logger.Warn("Failed to return CID to pool",
+				logfields.CIDName, cidName,
```

Comment:
If this is hit, will we have enough context on the labels map for this CID?

Reply:
I agree. More info would be better here for sure. I added labels to the log.
Thanks for the suggestion!